time_interp/include/time_interp.inc

```diff
@@ -653,81 +335,35 @@ character(len=*), intent(out), optional :: err_msg
   if( T>=Timelist(ie) ) then
      ! time is after the end of the portion of the time list within the requested period
      index1 = ie;   index2 = is
-     weight = real( (T-Timelist(ie))//(Period-(Timelist(ie)-Timelist(is))) )
+     weight = real((T-Timelist(ie))//(Period-(Timelist(ie)-Timelist(is))), FMS_TI_KIND_ )
   else if (T<Timelist(is)) then
      ! time is before the beginning of the portion of the time list within the requested period
      index1 = ie;   index2 = is
-     weight = real( 1.0-((Timelist(is)-T)//(Period-(Timelist(ie)-Timelist(is)))) )
+     weight = real( 1.0_kindl -((Timelist(is)-T)//(Period-(Timelist(ie)-Timelist(is)))), FMS_TI_KIND_ )
```

Comment:
wondering how the math works here, real number - timetype

Reply:
The `//` is actually its own operator (as opposed to `/` ) for real division, so by the time it gets to the subtraction it'll be real - real.

Reply:
got it